src/main/java/com/cloudogu/smeagol/repository/domain/BranchRepository.java

```diff
@@ -0,0 +1,16 @@
+package com.cloudogu.smeagol.repository.domain;
+
+/**
+ * Repository to query branches.
+ */
+public interface BranchRepository {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Annotate the "BranchRepository" interface with the @FunctionalInterface annotation [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS1609)